### PR TITLE
9712 - Search is showing different results

### PR DESCRIPTION
**Pull request by fessehaye**
# Description

Main problem is not the search showing different results, its that the products are being sorted differently when the search text is apart of the starting URL

See related issue for repro steps. You might need to use the prod database to test

Related PRs/issues: #9712

### CODE REVIEW

File: network-api/networkapi/templates/pages/buyersguide/catalog.html [hunk 1]
```diff
@@ -79,7 +79,7 @@
         <div class="tw-w-full tw-px-4 xlarge:tw-px-0 tw-flex tw-relative tw-flex-wrap medium:tw-flex-nowrap medium:tw-left-auto medium:tw-ml-0 medium:tw-static">
           <div class="tw-flex tw-items-end tw-space-x-2 tw-overflow-auto tw-pb-2 tw-no-scrollbar tw-touch-pan-x tw-w-full medium:tw-w-auto medium:w-auto subcategory-header">
 
-            <span id="product-filter-pni" class="tw-flex tw-cursor-pointer tw-text-gray-60 border tw-border-gray-20 tw-px-2 tw-py-1 tw-font-sans tw-rounded-3xl tw-font-normal tw-text-[12px] tw-leading-[1.3] hover:tw-border-blue-10 hover:tw-bg-blue-10">
```
Comment (fessehaye) [inline network-api/networkapi/templates/pages/buyersguide/catalog.html:82]: this change is just to solve this visual issue 
image

File: source/js/buyers-guide/search/history.js [hunk 2]
```diff
@@ -117,7 +117,12 @@ export function performInitialHistoryReplace(
       });
   }
 
-  Utils.sortProductCards();
+  // sorting priorities change when search text is applied
+  if (searchParameter) {
+    Utils.sortFilteredProducts();
+  } else {
+    Utils.sortProductCards();
+  }
```
Comment (mmmavis) [inline source/js/buyers-guide/search/history.js:125]: Hi @fessehaye , 

I don't know the specs and codebase well enough. Could you help clarify why we have two different sorting functions? I imagine the sorting logic would be the same whether it's a full product list or a filtered one. The only difference would be which project list we apply the same sorting function on.

Comment (fessehaye) [reply source/js/buyers-guide/search/history.js:125]: So `sortFilteredProducts()` applies how products should be sorted from last years update https://github.com/mozilla/foundation.mozilla.org/issues/7476

Where `sortProductCards()` applies what sort is applied from the pni-category dropdown that was added this years update.

So the sorting logic is different whether it's a full product list or a filtered one. This is a little confusing so we added it for discussion for the next PNI meeting!

Comment (mmmavis) [reply source/js/buyers-guide/search/history.js:125]: Got it! Thanks for the explanation.

**Review by mmmavis — APPROVED** [review]
R+!

**Pull request merged**